### PR TITLE
Img tag attrs

### LINKED ISSUE #7: title attribute being parsed with twemoji.js
I'm using twemoji.js on the front end for some preview features. I'm using twemoji.rb for backend rendering. If I take these steps I run into issues:

  1. If I backend render an img tag with Twemoji.parse that includes something like the deafult title `title = ":heart_eyes:"`
  1.  edit that text (which includes the `` tag) in a wysiwyg editor on the front end after backend rendering
  1. `twemoji.parse` in the frontend for preview the title attribute value is replaced for another img tag

:(

We can fix this a number of ways in our own app, but if we made one of three possible easy modifications to twemoji.rb it would be a non-issue. 

  1. change title to be the shortname without the `:` so `heart_eyes` instead of `:heart_eyes:`
  1. remove title attribute from default (adding when needed)
  1. make all the attributes in the image tag come from an options hash like `options[:default_img_attrs]` or make `options[:img_attrs]` that default.

I would be happy to code whatever solution you prefer.

Thanks!

### PR DESCRIPTION
**Pull request by bramswenson**
Ok, this one will fix #7 for me and adds total control over each img tags attributes with a filtering.

### CODE REVIEW

**Comment by bramswenson** [comment]
I will need to add some documentation if you all are good with this one.

File: lib/twemoji.rb [hunk 1]
```diff
@@ -230,4 +238,17 @@ def self.emoji_url(name)
...
       end
     end
+
+    # Returns a string of attributes suitable for use in and html tag
+    #
+    # @param hash [Hash] Hash of attributes.
+    # @return [String] String of attributes suitable for use in tag
+    # @private
+    def self.hash_to_attributes(hash)
+      hash.inject('') do |attributes, (attr, value)|
```
Comment (winston) [inline lib/twemoji.rb]: Would you consider a more succint version:

```
def self.hash_to_attributes(hash)
  hash.map { |attr_value|  attr_value.join('=') }.join(' ')
end
```

File: lib/twemoji.rb [hunk 2]
```diff
@@ -206,12 +204,22 @@ def self.filter_emojis(content)
...
-      if options[:img_attr]
-        "<img class='#{options[:class_name]}' draggable='false' title='#{name}' alt='#{unicode}' #{options[:img_attr]} src='#{emoji_url(name)}'>"
-      else
-        "<img class='#{options[:class_name]}' draggable='false' title='#{name}' alt='#{unicode}' src='#{emoji_url(name)}'>"
+      # set default attributes and override from img_attr
+      attributes = {
+        draggable: 'false',
+        title:     name,
+        alt:       render_unicode(name),
+        src:       emoji_url(name)
```
Comment (winston) [inline lib/twemoji.rb]: I am thinking if all the attrs above should be overidable? For example.. `src` would you ever want to override that?

File: test/twemoji_test.rb [hunk 3]
```diff
@@ -75,21 +75,35 @@ def test_parse_option_unsupport_file_ext_gif
...
+    shortname_filter = lambda { |name| name.gsub(':', '') }
+    img_attr = {
+      style: 'height: 1.3em;',
+      title: shortname_filter,
+      alt:   shortname_filter,
+    }
+
+    expected = "I like chocolate <img draggable='false' title='heart_eyes' alt='heart_eyes' src='https://twemoji.maxcdn.com/16x16/1f60d.png' style='height: 1.3em;' class='twemoji'>!"
+
+
```
Comment (winston) [inline test/twemoji_test.rb]: :scissors: empty line?

**Comment by winston** [comment]
I am good with the premise of the change. :smile: Thank you!

File: .gitignore [hunk 4]
```diff
@@ -12,3 +12,4 @@
 *.o
 *.a
 mkmf.log
+*.swp
```
Comment (JuanitoFatas) [inline .gitignore]: This should go to your [global gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

File: lib/twemoji.rb [hunk 5]
```diff
@@ -106,13 +105,12 @@ def self.parse(text, asset_root: "https://twemoji.maxcdn.com/",
                        file_ext:   ".png",
                        image_size: "16x16",
                        class_name: "emoji",
-                       img_attr: nil)
+                       img_attr: {})
```
Comment (JuanitoFatas) [inline lib/twemoji.rb]: ok so this old api is intended to pass all attributes in one string, now we should rename to `img_attrs`.

File: lib/twemoji.rb [hunk 6]
```diff
@@ -206,12 +204,22 @@ def self.filter_emojis(content)
...
-        "<img class='#{options[:class_name]}' draggable='false' title='#{name}' alt='#{unicode}' #{options[:img_attr]} src='#{emoji_url(name)}'>"
-      else
-        "<img class='#{options[:class_name]}' draggable='false' title='#{name}' alt='#{unicode}' src='#{emoji_url(name)}'>"
+      # set default attributes and override from img_attr
+      attributes = {
+        draggable: 'false',
+        title:     name,
+        alt:       render_unicode(name),
+        src:       emoji_url(name)
+      }.merge(options[:img_attr])
```
Comment (JuanitoFatas) [inline lib/twemoji.rb]: `merge!` is safe to use here.

Comment (winston) [inline lib/twemoji.rb]: Or we can do 

```
options[:img_attr].reverse_merge!(
  draggable: 'false',
  title:     name,
  alt:       render_unicode(name),
  src:       emoji_url(name)
)
```

Comment (JuanitoFatas) [inline lib/twemoji.rb]: :+1: But we did not include Active Support.

Comment (winston) [inline lib/twemoji.rb]: Oops. Ok. just use `merge!` then. :smile:

**Comment by bramswenson** [comment]
I'll get this one updated and resubmitted this week. Thanks for all the input! You guys do great code reviews!

**Comment by JuanitoFatas** [comment]
Superseded by #16.

**Pull request closed without merging**